config/cmake/cacheinit.cmake

```diff
@@ -40,6 +40,8 @@ set (HDF_TEST_EXPRESS "2" CACHE STRING "Control testing framework (0-3)" FORCE)
 
 set (HDF5_MINGW_STATIC_GCC_LIBS ON CACHE BOOL "Statically link libgcc/libstdc++" FORCE)
 
+set (CMAKE_DEBUG_POSTFIX "" CACHE STRING "Filename postfix for debug builds" FORCE)
```

Comment:
Just replace this with the whole thing:
  if(NOT CMAKE_DEBUG_POSTFIX)
    if (WIN32)
      set (CMAKE_DEBUG_POSTFIX "_D")
    else ()
      set (CMAKE_DEBUG_POSTFIX "_debug")
    endif ()
  endif ()